messaginginapp/src/main/java/io/customer/messaginginapp/gist/data/listeners/Queue.kt

```diff
@@ -141,6 +145,19 @@ class Queue : GistListener {
         }
     }
 
+    private fun updatePollingInterval(headers: Headers) {
+        headers["X-Gist-Queue-Polling-Interval"]?.let { pollingInterval ->
+            if (pollingInterval.toInt() > 0) {
+                val newPollingInterval = (pollingInterval.toInt() * 1000).toLong()
+                if (newPollingInterval != GistSdk.pollInterval) {
+                    GistSdk.pollInterval = newPollingInterval
+                    GistSdk.observeMessagesForUser(true)
+                    Log.i(GIST_TAG, "Polling interval changed to: ${pollingInterval.toInt()} seconds")
+                }
+            }
```

Comment:
```suggestion
        headers["X-Gist-Queue-Polling-Interval"]?.toIntOrNull()?.let { pollingIntervalSeconds ->
            if (pollingInterval.toInt() > 0) {
                val pollingIntervalMilliseconds = (pollingIntervalSeconds * 1000).toLong()
                if (pollingIntervalMilliseconds != GistSdk.pollInterval) {
                    GistSdk.pollInterval = pollingIntervalMilliseconds
                    
                    // Queue check fetches messages again and could result in infinite loop. 
                    GistSdk.observeMessagesForUser(skipQueueCheck = true)
                    Log.i(GIST_TAG, "Polling interval changed to: ${pollingIntervalSeconds} seconds")
                }
            }
```

I have a couple suggestions that makes the code easier for me to follow. Nothing major though. 

* Labeling the variables with "seconds" and "milliseconds" made code faster to read. 
* Added comment on why `GistSdk.observeMessagesForUser` is skipping queue check. Maybe my comment is incorrect? I left the comment based on my understanding of the code after reading. 
* Moved the `pollingInterval.toInt()` to the `let {}` statement, to prevent having to call it multiple times.

Reply:
Agreed, I'll update the PR and implement some other minor changes.